core/src/main/groovy/com/novoda/gradle/release/ReleasePlugin.groovy

```diff
@@ -49,26 +49,25 @@ class ReleasePlugin implements Plugin<Project> {
     void attachArtifacts(PublishExtension extension, Project project) {
         if (project.plugins.hasPlugin('com.android.library')) {
             project.android.libraryVariants.all { variant ->
-                def artifactId = extension.artifactId;
-                addArtifact(project, variant.name, artifactId, new AndroidArtifacts(variant))
+                addArtifact(project, variant.name as String, extension, new AndroidArtifacts(variant))
```

Comment:
Do you really need `as String` here? I think it is clear that `variant.name` is a String

Reply:
For us it is clear. But not for the IDE.

We have a general problem here:
All of the `android` stuff is unknown since we don't have the android gradle plugin as dependencies. So variant is currently just a "object".

I don't know of it make sense to include the android gradle plugin as dependencie since it needed the google() repository. But that means if a java project applies our plugin it has to define the google() repo.. which is a little bit dump IMO...

How do we want proceed here? 🤔

Reply:
Second part is another discussion. But for the first one, what do you think is the benefit of using `as String` here in terms of IDE support. I don't think it changes anything at all.

Reply:
Sure, it is related and not another discussion.
Because `variant` isn't a concrete class or something... It is just unknown for the IDE.
But I think I found a solution. If we use the android gradle plugin as a compileOnly dependency it should work with IDE support :)